kong/llm/drivers/cohere.lua

```diff
@@ -12,102 +12,67 @@ local table_new = require("table.new")
 local DRIVER_NAME = "cohere"
 --
 
-local transformers_to = {
-  ["llm/v1/chat"] = function(request_table, model)
-    request_table.model = model.name
+local function merge_fields(request_table, model)
+  request_table.temperature = request_table.temperature or model.options.temperature
+  request_table.max_tokens = request_table.max_tokens or model.options.max_tokens
+  request_table.truncate = request_table.truncate or "END"
+  request_table.return_likelihoods = request_table.return_likelihoods or "NONE"
+  request_table.p = request_table.top_p or model.options.top_p
+  request_table.k = request_table.top_k or model.options.top_k
+
+  return request_table
+end
 
-    if request_table.prompt and request_table.messages then
-      return kong.response.exit(400, "cannot run a 'prompt' and a history of 'messages' at the same time - refer to schema")
-  
-    elseif request_table.messages then
-      -- we have to move all BUT THE LAST message into "chat_history" array
-      -- and move the LAST message (from 'user') into "message" string
-      if #request_table.messages > 1 then
-        local chat_history = table_new(#request_table.messages - 1, 0)
-        for i, v in ipairs(request_table.messages) do
-          -- if this is the last message prompt, don't add to history
-          if i < #request_table.messages then
-            local role
-            if v.role == "assistant" or v.role == "CHATBOT" then
-              role = "CHATBOT"
-            else
-              role = "USER"
-            end
-    
-            chat_history[i] = {
-              role = role,
-              message = v.content,
-            }
+local function handle_all(request_table, model)
+  request_table.model = model.name
+
+  if request_table.prompt and request_table.messages then
+    return kong.response.exit(400, "cannot run a 'prompt' and a history of 'messages' at the same time - refer to schema")
+
+  elseif request_table.messages then
+    -- we have to move all BUT THE LAST message into "chat_history" array
+    -- and move the LAST message (from 'user') into "message" string
+    if #request_table.messages > 1 then
+      local chat_history = table_new(#request_table.messages - 1, 0)
+      for i, v in ipairs(request_table.messages) do
```

Comment:
A numeric for loop from `1` to `#request_table.messages - 1` would be better than `ipairs` here.